File: Algorithms/Sortings/JavaScript/bubble-sort.js
function bubbleSort(array) {
  for (let i = 0; i < array.length; i++) {
    for (let j = 0; j < array.length; j++) {
      if (array[j] > array[j + 1]) {
        let temp = array[j];
        array[j] = array[j + 1];
        array[j + 1] = temp;
      }
    }
  }
  return array;
}

console.log(
  bubbleSort([
    120, 28, 23, 24, 48, 454, 1, 402, 31, 44, 50, 420, 3, 15, 14, 16, 35, 27,
    111, 30, 11, 6, 230, 37, 50, 8, 26, 505, 1588, 2298,
  ])
);

File: Algorithms/Sortings/Java/bubble_sort.java
package com.gmail.tsa;
import java.util.Arrays;

public class Main {
	public static void main(String[]) {
		int[] array = new int[] {5, 0, -2, 7, 3};
		int sortedIndex = array.length - 1;
		int numberOfSwap = 1;

		while(numberOfSwap > 0) {
			numberOfSwap = 0
			for(int i = 0; i < sortedIndex; i++) {
				if(array[i] > array[i + 1]) {
					int temp = array[i];
					array[i] = array[i + 1];
					array[i + 1] = temp;
					numberOfSwap += 1;
				}
			}
			sortedIndex -= 1;
		}

		System.out.println(Arrays.toString(array));
	}
}

void bubbleSort(int[] arr) {
	int length = arr.length;
	while(length != 0) {
		int max_index = 0;
		for(int i = 1; i < length; i++) {
			if(arr[i - 1] > arr[i]) {
				int temp = arr[i - 1];
				arr[i - 1] = arr[i];
				arr[i] = temp;

				max_index = i;
			}
		}
		length = max_index;
	}
}